src/manage_migrators.c

```diff
@@ -1564,6 +1564,38 @@ migrate_220_to_221 ()
   return 0;
 }
 
+/**
+ * @brief Migrate the database from version 221 to version 222.
+ *
+ * @return 0 success, -1 error.
+ */
+int
+migrate_221_to_222 ()
+{
+  sql_begin_immediate ();
+
+  /* Ensure that the database is currently version 221. */
+
+  if (manage_db_version () != 221)
+    {
+      sql_rollback ();
+      return -1;
+    }
+
+  /* Update the database. */
+
+  /* Remove permissions on the remove command MODIFY_REPORT */
```

Comment:
This needs to remove any tags on the permissions too.  Something like:

    sql ("DELETE FROM tag_resources"
         " WHERE location = %i"
         " AND resource_type = 'permission'"
         " AND resource_uuid = (SELECT uuid FROM permissions WHERE name = 'modify_report';",
         LOCATION_TABLE);

Same for LOCATION_TRASH.  Probably need 4 cases actually, for tag_resources_trash.

Reply:
I don't see any "orphan cleanup" of tag_resources.

I don't see how you are going to simplify this much.  You still need to check that the resource_type, location and resource_uuid refer to a resource that does not exist.

Perhaps you means to drop " WHERE name = 'modify_report'" from the query?  I don't think that is so clear in a migrator that is about modify_report, but I don't feel strongly about it.

If you want to add the orphan cleanup at startup I think it will be much more work because you have to consider all the types.

Reply:
I looked at this again.  We haven't cleaned up the tag_resource in past migrators, and it's not very likely that someone tagged a permission, so I think it's OK.  Even if someone tagged the permission it'll just become an orphaned tag.